### PR TITLE
[Kernel] Return `tags` in `Scan.getScanFiles()` output

### PR DESCRIPTION
**Pull request by horizonzy**
Now, the file scan result didn't contain a tags field. In our case, we define some custom properties in the tags filed, so we want to filter the AddFile according to the custom tags.

### CODE REVIEW

**Comment by horizonzy** [comment]
@vkorukanti /cc

File: kernel/kernel-api/src/main/java/io/delta/kernel/internal/ScanImpl.java [hunk 1]
```diff
@@ -94,7 +94,7 @@ public ScanImpl(
      */
     @Override
     public CloseableIterator<FilteredColumnarBatch> getScanFiles(Engine engine) {
-        return getScanFiles(engine, false);
```
Comment (vkorukanti) [inline kernel/kernel-api/src/main/java/io/delta/kernel/internal/ScanImpl.java:97]: Let's make reading the tags by default. Most cases the contents are empty, so shouldn't add extra overhead for reads.

**Review by vkorukanti — CHANGES_REQUESTED** [review]
@horizonzy Thanks for the PR. One change request: always reads the `tags`,  so that the `Scan.getScanFiles()` always has the `tags` for each returned row. That should simplify this PR.

And also please update the `Scan.getScanFiles()` javadoc to include the new columns `tags`.

**Comment by horizonzy** [comment]
> @horizonzy Thanks for the PR. One change request: always reads the `tags`, so that the `Scan.getScanFiles()` always has the `tags` for each returned row. That should simplify this PR.
> 
> And also please update the `Scan.getScanFiles()` javadoc to include the new columns `tags`.

The method `Scan.getScanFiles()` doc links the SCAN_FILE_SCHEMA_WITH_STATS, I think we needn't mark the changes for the method. We can mark the change in the next release note.

File: kernel/kernel-api/src/main/java/io/delta/kernel/internal/actions/AddFile.java [hunk 2]
```diff
@@ -55,20 +55,16 @@ public class AddFile {
         .add("size", LongType.LONG, false /* nullable*/)
         .add("modificationTime", LongType.LONG, false /* nullable*/)
         .add("dataChange", BooleanType.BOOLEAN, false /* nullable*/)
-        .add("deletionVector", DeletionVectorDescriptor.READ_SCHEMA, true /* nullable */);
+        .add("deletionVector", DeletionVectorDescriptor.READ_SCHEMA, true /* nullable */)
+        .add("tags", new MapType(StringType.STRING, StringType.STRING, true), true);
```
Comment (vkorukanti) [inline kernel/kernel-api/src/main/java/io/delta/kernel/internal/actions/AddFile.java:59]:
```suggestion
        .add("tags",
                new MapType(StringType.STRING, StringType.STRING, true /* valueContainsNull */),
                true /* nullable */);
```

File: kernel/kernel-api/src/main/java/io/delta/kernel/utils/FileStatus.java [hunk 3]
```diff
@@ -68,6 +76,24 @@ public long getModificationTime() {
...
+
+    /**
+     * Get the stats of the file.
+     *
+     * @return the file stats
+     */
+    public String getStats() {
+        return stats;
+    }
+
```
Comment (vkorukanti) [inline kernel/kernel-api/src/main/java/io/delta/kernel/utils/FileStatus.java:96]: `FileStatus` is a public API and we can't have these fields in `FileStatus`. If you need an object with file info, stats, and tags, you can create one in your connector and write a utility method similar to the `InternalScanFileUtils.getAddFileStatus`.

**Review by vkorukanti — CHANGES_REQUESTED** [review]
Looks good.

Fews things:
* Please update the javadoc [here](https://github.com/delta-io/delta/blob/master/kernel/kernel-api/src/main/java/io/delta/kernel/Scan.java#L68) for `Scan.getScanFileInfo`. Basically it contains the schema of the scan file rows returned. You can just add
```
<li>name: {@code tags}, type: {@code map(string, string)}, description: Map containing metadata about the scan file.</li>
```

* Remove `stats` and `tags` from the `FileStatus`. They don't belong there. I commented on the PR about the alternative.

* A simple test. You can add a Delta table which contains tags for AddFile. And verify that `Scan.getScanFiles()` returns the tags. You can add this to `ScanSuite.scala`. 

Let me know if you any questions.

**Review by vkorukanti — APPROVED** [review]

**Pull request merged**